src/bl.cpp

```diff
@@ -1852,7 +1853,8 @@ static void log_POST(char *log_buffer, size_t size)
   {
     Log_info("Was unable to send log to API; saving locally for later.");
     // log not send
-    store_log(log_buffer, size, preferences);
+    PreferencesPersistence preferencesPersistence(preferences);
```

Comment:
Starting small, for now `PreferencesPersistence` is used in just a few places. Down the road I imagine having a single instance replacing the app's use of `preferences`.